### PR TITLE
symbolic shape inference improvements

### PR DESCRIPTION
**Pull request by tianleiwu**
**Description**: 
(1) Add EmbedLayerNormalization in shape inference
(2) Do not call onnx shape inference for contrib ops that defined inference function.
(3) Pass initializers for onnx shape inference that for Unsqueeze, so that we can remove custom shape inference code for the operator (as in https://github.com/microsoft/onnxruntime/pull/7560)
(4) Update unit test script so that we can run test without compiling and installing onnxruntime package. It helps developing.
(5) Fix a warning in Attention shape inference by converting float to int.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

When model has contrib op, symbolic shape inference output some warnings:
```
Warning: Unsupported operator LayerNormalization. No schema registered for this operator.
```
Need avoid calling onnx shape inference for contrib operators (some contrib ops are defined in onnx domain but they shall not).

### CODE REVIEW

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 1]
```diff
@@ -326,18 +327,26 @@ def _update_computed_dims(self, new_sympy_shape):
                         self.symbolic_dims_[str(new_dim)] = new_dim
 
     def _onnx_infer_single_node(self, node):
-        # skip onnx shape inference for some ops, as they are handled in _infer_*
-        skip_infer = node.op_type in ['If', 'Loop', 'Scan', 'SplitToSequence', 'ZipMap']
+        # skip onnx shape inference for some ops, as they are handled in _infer_* or it is not in onnx domain
+        skip_infer = node.domain in ['com.microsoft'] \
```
Comment (tianleiwu) [inline onnxruntime/python/tools/symbolic_shape_infer.py:331]: It seems skip inference of MS domain will cause training failure.

Comment (tianleiwu) [reply onnxruntime/python/tools/symbolic_shape_infer.py]: Resolved: only skip contrib ops that defines _infer_*

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 2]
```diff
@@ -326,18 +327,26 @@ def _update_computed_dims(self, new_sympy_shape):
...
+
+            # Only pass initializers with impact on shape inference result for performance.
+            initializers = [self.initializers_[name] for name in node.input
+                            if name in self.initializers_] if need_initializers else []
+
             # run single node inference with self.known_vi_ shapes
-            # note that inference rely on initializer values is not handled
-            # as we don't copy initializer weights to tmp_graph for inference speed purpose
             tmp_graph = helper.make_graph(
                 [node], 'tmp', [self.known_vi_[i] for i in node.input if i],
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply onnxruntime/python/tools/symbolic_shape_infer.py:352]: resolved: add a check that opset version >= 9 and initializer is not in graph input.

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 3]
```diff
@@ -327,17 +330,32 @@ def _update_computed_dims(self, new_sympy_shape):
...
+        ]
+
         if not skip_infer:
+            # Only pass initializers that satisfy the following condition:
+            # (1) Operator need value of some input for shape inference.
+            #     For example, Unsqueeze in opset 13 uses the axes input to calculate shape of output.
+            # (2) opset version >= 9. In older version, initializer is required in graph input by onnx spec.
+            # (3) The initializer is not in graph input. The means the node input is "constant" in inference.
+            initializers = []
+            if (get_opset(self.out_mp_) >= 9) and node.op_type in ['Unsqueeze']:
```
Comment (KeDengMS) [inline onnxruntime/python/tools/symbolic_shape_infer.py:345]: >Unsqueeze [](http://example.com/codeflow?start=68&length=9)

I think we may add more known tensor ops in opset 13 that changed from attribute to axes:
Expand, Slice, Squeeze,

**Review by KeDengMS — APPROVED** [review]
:shipit:

**Pull request merged**